ampd/src/sui/verifier.rs

```diff
@@ -0,0 +1,194 @@
+use move_core_types::language_storage::StructTag;
+use serde::Deserialize;
+use sui_json_rpc_types::{SuiEvent, SuiTransactionBlockResponse};
+use sui_types::base_types::{ObjectID, SuiAddress};
+
+use crate::handlers::sui_verify_msg::Message;
+use crate::types::Hash;
+
+const CONTRACT_CALL_EVENT: &str = "gateway::ContractCall";
```

Comment:
Is `gateway` something hard-coded in the sui contract?

Reply:
ya, `gateway` is the module name. the event type is in form of `<package id>::<module name>::<event type>`

Reply:
Can you add a comment explaining it?